### ISSUE TITLE
Fix epydoc warnings in twisted.words

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8536]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8536|
|Type|enhancement|
|Created|2016-06-29 16:37:20Z|
|Branch|https://github.com/twisted/twisted/tree/8536-rodrigc-doc|

During building of apidocs, there are some epydoc warnings.
These should be fixed.

<details><summary>Searchable metadata</summary>

```
trac-id__8536 8536
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__None None
branch__8536_rodrigc_doc 8536-rodrigc-doc
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__None None
time__1467218240510837 1467218240510837
changetime__1467720932919797 1467720932919797
version__None None
owner__adiroiban adiroiban
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/270

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Changes look good, but I think that we should document the xmlstream public constants

I don't see them documented here, and this is why the L{} references are failing... but they should be documented.

So instead of using C{} we should keep using L{} for those constants and document them in the API.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_4"></a>|@ralphm commented|
|-|-|

Replying to [adiroiban](#note_3):
> So instead of using C{} we should keep using L{} for those constants and document them in the API.

Agreed. Just because they currently happen to raise a warning, doesn't mean we should replace the semantic usage of `L{}` with the generic `C{}`.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @adiroiban|
|-|-|

Let me try to see how I can add those docs ... and then I hope that I can get some help from Ralph to make sure we have valid documentation.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban set owner to @ralphm|
|-|-|

Ok.

```
./bin/admin/build-apidocs . apidocs/ | grep words
```

is clean on my branch.

Ralph, please check that docstring are ok.

I did a quick and dirty job to add docstrings as I was aiming to get zero errors rather than accurate docstrings.

Happy to see that words is pydoctor error free :)

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_7"></a>|@ralphm set owner to @adiroiban|
|-|-|

There were some minor typos and other remarks that I commented on in the PR. Please resolve those and then please merge.

Thanks!

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|GitHub set status to `closed`|
|-|-|

In changeset e79df111bd31fdc93f3ac568c16797cfccf46e43
```
#!CommitTicketReference repository="" revision="e79df111bd31fdc93f3ac568c16797cfccf46e43"
Merge 8536-rodrigc-doc: Fix pydoctor errors in twisted.words.

Author: rodrigc, adiroiban
Reviewer: adiroiban, ralphm
Fixes: #8536
```
